errai-bus/src/main/java/org/jboss/errai/bus/client/framework/ClientMessageBusImpl.java

```diff
@@ -948,25 +948,44 @@ private void sendAllDeferred() {
     final List<Message> highPriority = new ArrayList<>();
     for (final Message message : new ArrayList<>(deferredMessages)) {
       if (message.hasPart(MessageParts.PriorityProcessing)) {
-        if (remotes.containsKey(message.getSubject()))
+        if (remotes.containsKey(message.getSubject())) {
           highPriority.add(message);
-        deferredMessages.remove(message);
+          deferredMessages.remove(message);
+        }
       }
     }
 
     final List<Message> lowPriority = new ArrayList<>();
-    do {
+    for(Message m : deferredMessages) {
       for (final Message message : new ArrayList<>(deferredMessages)) {
-        if (remotes.containsKey(message.getSubject()))
+        if (remotes.containsKey(message.getSubject())) {
           lowPriority.add(message);
-        deferredMessages.remove(message);
+          deferredMessages.remove(message);
+        }
       }
     }
-    while (!deferredMessages.isEmpty());
 
-    transportHandler.transmit(highPriority);
-    transportHandler.transmit(lowPriority);
-    deferredMessages.clear();
+    try {
+      transportHandler.transmit(highPriority);
+      transportHandler.transmit(lowPriority);
+
+      for (final Message message : deferredMessages) {
+        String subject = message.getSubject();
+        if (!localSubscriptions.containsKey(subject) ||
```

Comment:
The condition here is not quite right. What I had intended was:

    !(localSubscriptions.containsKey(subject)
      || subscriptions.containsKey(subject)
      || shadowSubscriptions(subject))

Or in English: There is no subscription in any of local subscriptions, subscriptions, or shadow subscriptions.

I think fixing this condition will fix the remainder of the test failures.

Reply:
👍